scirpy/_tools/_diversity.py

```diff
@@ -65,8 +91,24 @@ def _shannon_entropy(freq):
     diversity = dict()
     for k in sorted(ir_obs[groupby].unique()):
         tmp_counts = clono_counts.loc[clono_counts[groupby] == k, "count"].values
-        tmp_freqs = tmp_counts / np.sum(tmp_counts)
-        diversity[k] = _shannon_entropy(tmp_freqs)
+        
+        if method in ["normalized shannon entropy", "D50"]:
+            # calculate frequencies for these two metrics
+            tmp_freqs = tmp_counts / np.sum(tmp_counts)
+            np.testing.assert_almost_equal(np.sum(tmp_freqs), 1)
+
+            if method == "normalized shannon entropy":
+                diversity[k] = _shannon_entropy(tmp_freqs)
+            else:
+                diversity[k] = _dxx(tmp_freqs, perc=0.5)
+        
+        elif method in skbio.diversity.get_alpha_diversity_metrics():
+            # skbio.diversity takes count vectors as input
+            diversity[k] = skbio.diversity.alpha_diversity(method, tmp_counts)
+        
+        else:
+            # raise if method is not supported
+            raise ValueError(f"{method} diversity metric is not supported")
```

Comment:
What do you think of, additionally, adding the option to pass a callback function to `method`? Like that users could bring their own diversity metrics.

Reply:
This is a good point. However, should we provide example code and specify whether they use freqs or counts?

Reply:
I think it's ok to use counts always, as its trivial to calculate frequencies from that. In that case I'd also adapt the two internal metrics to accept counts.

Reply:
Sounds good. I have modified `_shannon_entropy` and `_dxx` to take counts instead and added support for custom metric function. Please let me know what you think. Thank you!

Reply:
Hi @grst, I wanted to follow up with you. Please let me know if you have further suggestions. Thanks!